examples/lora/sensor-lorawan/main.cc

```diff
@@ -0,0 +1,107 @@
+/*
+   RadioLib Non-Arduino Tock Library LoRaWAN test application
+
+   Licensed under the MIT or Apache License
+
+   Copyright (c) 2023 Alistair Francis <alistair@alistair23.me>
+ */
+
+// include the library
+#include <RadioLib.h>
+
+// include the hardware abstraction layer
+#include "libtockHal.h"
+
+// Include some libtock-c helpers
+#include <libtock-sync/sensors/humidity.h>
+#include <libtock-sync/sensors/temperature.h>
+
+#include "CayenneLPP.h"
+
+// To get this working copy radioConfig_example.h to radioConfig.h
+// and then modify it to match the LoRaWAN gateway settings.
+#ifdef RADIO_CONFIG_CI
+#include "radioConfig_example.h"
+#else
+#include "radioConfig.h"
+#endif
+
+#define MAX_SIZE 10
+
+// the entry point for the program
+int main(void) {
+  CayenneLPP Payload(MAX_SIZE);
+
+  printf("[SX1261] Initialising Radio ... \n");
+
+  // create a new instance of the HAL class
+  TockHal* hal = new TockHal();
```

Comment:
```suggestion
  TockRadioLibHal* hal = new TockRadioLibHal();
```

Or I suppose since this is C++, even better would be to wrap everything in a `namespace RadioLib` (or `RadioLibHal` or `TockRadioLib::{Hal, Module}`, etc — I might like this last option best but don't feel strongly). The same issue applies for the "`Module`" below — these are generic terms that make composition really hard.

[I know this comes from upstream, so the comment doesn't apply to this PR directly per se, but IIUC, you're also driving the Tock interface in RadioLib, so hopefully this is an easy enough change to push there and propagate here via submodule hash bump]